From 93c2ddb82be5ea15b2ae980bde97b5d1d9ed6075 Mon Sep 17 00:00:00 2001
From: Steve Nicholson <ssteve@mac.com>
Date: Thu, 2 Jun 2016 12:31:21 -0700
Subject: [PATCH 1/3] Fix installation directory. Without this line, the lib
 files don't have the correct path embedded and can't be loaded in OS X.

---
 cmake/OpenCVModule.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/OpenCVModule.cmake b/cmake/OpenCVModule.cmake
index bd3286c..24c7054 100644
--- a/cmake/OpenCVModule.cmake
+++ b/cmake/OpenCVModule.cmake
@@ -833,6 +833,7 @@ macro(_ocv_create_module)
     COMPILE_PDB_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     LIBRARY_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
+    INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/lib
   )

   # For dynamic link numbering convenions

From a47a5ca056d70c637f66bd0e3990ea2de38f444a Mon Sep 17 00:00:00 2001
From: Steve Nicholson <ssteve@mac.com>
Date: Thu, 9 Jun 2016 12:25:07 -0700
Subject: [PATCH 2/3] Set CMAKE_INSTALL_NAME_DIR when CMAKE_INSTALL_PREFIX
 isn't default value of "/usr/local" in a Unix/macOS/CygWin installation. If
 this isn't set, it causes one of two problems: 1) If OpenCV libraries are
 already installed in /usr/local, these will be loaded instead of the versions
 in the non-default location or 2) If OpenCV libraries aren't already
 installed in /usr/local, the program will not run because the libraries in
 the non-default location won't be found.

---
 cmake/OpenCVModule.cmake | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cmake/OpenCVModule.cmake b/cmake/OpenCVModule.cmake
index bd3286c..df92142 100644
--- a/cmake/OpenCVModule.cmake
+++ b/cmake/OpenCVModule.cmake
@@ -75,6 +75,12 @@ set(OPENCV_MODULES_DISABLED_AUTO  "" CACHE INTERNAL "List of OpenCV modules impl
 set(OPENCV_MODULES_DISABLED_FORCE "" CACHE INTERNAL "List of OpenCV modules which can not be build in current configuration")
 unset(OPENCV_WORLD_MODULES CACHE)

+# set CMAKE_INSTALL_NAME_DIR if CMAKE_INSTALL_PREFIX isn't default value of "/usr/local"
+if(UNIX AND NOT ${CMAKE_INSTALL_PREFIX} STREQUAL "/usr/local")
+  set(CMAKE_INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/lib)
+#  message ("setting CMAKE_INSTALL_NAME_DIR: ${CMAKE_INSTALL_NAME_DIR}")
+endif()
+
 # adds dependencies to OpenCV module
 # Usage:
 #   add_dependencies(opencv_<name> [REQUIRED] [<list of dependencies>] [OPTIONAL <list of modules>] [WRAP <list of wrappers>])

From e5ed50176a0167cd122b03e3e3dd4af0975bf7e8 Mon Sep 17 00:00:00 2001
From: Steve Nicholson <ssteve@mac.com>
Date: Thu, 9 Jun 2016 12:54:18 -0700
Subject: [PATCH 3/3] Remove older change

---
 cmake/OpenCVModule.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/OpenCVModule.cmake b/cmake/OpenCVModule.cmake
index 537f784..df92142 100644
--- a/cmake/OpenCVModule.cmake
+++ b/cmake/OpenCVModule.cmake
@@ -839,7 +839,6 @@ macro(_ocv_create_module)
     COMPILE_PDB_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     LIBRARY_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-    INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/lib
   )

   # For dynamic link numbering convenions
